internal/component/discovery/discovery.go

```diff
@@ -225,10 +225,19 @@ func (c *Component) runDiscovery(ctx context.Context, d DiscovererWithMetrics) {
 				send()
 				haveUpdates = false
 			}
-		case <-ctx.Done():
+		// If Run has exited, return.
```

Comment:
@thampiotr @wildum An alternative solution could be to modify the `case` on line 223 to send empty targets on the first tick after the component has started. This would clear the cached targets in cases when no targets were discovered, and it'd make sure we minimise the number of times we send an empty list of targets to downstream component. 

However, I'm not sure if we're guaranteed to discover targets within the first tick when starting a new component. It's possible that no targets were discovered yet, or that the list of targets is still incomplete.

In the case of an incomplete list of targets, if the list of discovered targets is small, I'm not sure if the gain in this solution is much better than simply sending an empty list of targets.